pkg/controller/job/job_controller.go

```diff
@@ -838,8 +838,11 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 	var manageJobErr error
 
 	exceedsBackoffLimit := jobCtx.failed > *job.Spec.BackoffLimit
+	jobCtx.finishedCondition = successCriteriaMetCondition(&job)
 
-	if feature.DefaultFeatureGate.Enabled(features.JobPodFailurePolicy) {
+	// Given that the Job already has the SuccessCriteriaMet condition, the termination condition already had confirmed in another cycle.
+	// So, the job-controller evaluates the podFailurePolicy only when the Job doesn't have the SuccessCriteriaMet condition.
+	if !isSuccessCriteriaMetCondition(jobCtx.finishedCondition) && feature.DefaultFeatureGate.Enabled(features.JobPodFailurePolicy) {
```

Comment:
```suggestion
	if jobCtx.finishedCondition == nil && feature.DefaultFeatureGate.Enabled(features.JobPodFailurePolicy) {
```
nit: At this point SuccessCriteriaMet condition would be the only option, so checking `jobCtx.finishedCondition == nil ` does the same, and I find it easier to read.

Reply:
I don't see such a need. The block of code tries to set `jobCtx.finishedCondition` based on JobPodFailurePolicy, so we only check this feature gate.

Reply:
I see. I'm ok with the replacement of `jobCtx.finished == nil`.

Reply:
Done.